pkg/scheduler/plugins/elasticquota/core/quota_operator.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	"encoding/json"
+
+	"sigs.k8s.io/scheduler-plugins/pkg/apis/scheduling/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/koordinator-sh/koordinator/apis/extension"
+)
+
+//"system-group" means that the "min" of this quota group is infinite, and its request will never be bound by the quota.
+// It is usually used for system level pods. When the scheduler starts, the "system-group" will be created by default not only
+// in scheduler memory, but also try to create the quota group crd. Its "min" and "max" are INT_MAX. At the same time, its
+//"min" will not be reduced in proportion to the above process. The real available total resource of normal quota groups is
+//the cluster total resource minus the "used" of the "sys-group".
+
+// We also need to introduce the concept of "default-group". If the pod cannot find a matching quota group, it will be
+// matched to the "default-group". the "default-group" will be created by default not only in scheduler memory, but also try
+// to create the quota group crd. Its "min" and "max" has default value, users can modify them on demand.
+
+//RootQuotaName means quotaTree's root\head.
+const (
+	SystemQuotaName  QuotaNameType = "system"
+	RootQuotaName    QuotaNameType = "root"
+	DefaultQuotaName QuotaNameType = "default"
+)
+
+type QuotaNameType string
+
+func GetParentQuotaInfo(quota *v1alpha1.ElasticQuota) (isParent bool, parentName QuotaNameType) {
+	isParent = IsParentQuota(quota)
+
+	parentName = GetParentQuotaName(quota)
+
+	return isParent, parentName
+}
+
+func GetParentQuotaName(quota *v1alpha1.ElasticQuota) QuotaNameType {
+	parentName, ok := quota.Labels[extension.LabelQuotaParent]
+	if ok && len(parentName) > 0 {
+		return QuotaNameType(parentName)
+	}
+
+	return RootQuotaName //default return RootQuotaName
+}
+
+func IsParentQuota(quota *v1alpha1.ElasticQuota) bool {
+	return quota.Labels[extension.LabelQuotaIsParent] == "true" //default return false
+}
+
+func IsAllowLentResource(quota *v1alpha1.ElasticQuota) bool {
+	return quota.Labels[extension.LabelAllowLentResource] != "false" //default return true
+}
+
+func GetSharedWeight(quota *v1alpha1.ElasticQuota) corev1.ResourceList {
```

Comment:
The functions in this file seem more suitable to be placed in the directory `apis/extension`, and consider creating a file named `elastic_quota.go` in the directory `apis/extensions` specifically to define the protocol and helper functions of ElasticQuota.